src/util/ByteOrder.hxx

```diff
@@ -516,7 +517,55 @@ public:
 			x = ByteSwap64(x);
 		return x;
 	}
-};
+} __attribute__((packed));
 
 static_assert(sizeof(PackedLE64) == sizeof(uint64_t), "Wrong size");
 static_assert(alignof(PackedLE64) == 1, "Wrong alignment");
+
+class PackedFloatLE
+{
+  union Value
+  {
+    uint32_t unsigned_int = 0;
+    float floating_point;
+  } __attribute__((packed)) value;
+
+public:
+  PackedFloatLE() = default;
+
+  PackedFloatLE(float f)
+  {
+    value.floating_point = f;
+    if (IsBigEndian()) {
+      value.unsigned_int = ByteSwap32(value.unsigned_int);
+    }
+  }
+} __attribute__((packed));
+
+static_assert(std::numeric_limits<float>::is_iec559);
```

Comment:
Why this? I know I suggested doing that, but the next guy won't understand it. And you didn't even document this requirement in the class's documentation. Oh, there's no API documentation at all!